src/suggesetor/VerseEditorSuggestor.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  Editor,
+  EditorPosition,
+  EditorSuggest,
+  EditorSuggestContext,
+  EditorSuggestTriggerInfo,
+  TFile,
+} from "obsidian";
+import BibleReferencePlugin from "../main";
+import { VerseTypoCheck } from "../VerseTypoCheck";
+import { VerseSuggesting } from "../VerseSuggesting";
+import { BibleReferencePluginSettings } from "../data/constants";
+import { getSuggestionsFromQuery } from "./VerseSuggestor";
+
+/**
+ * Extend the EditorSuggest to suggest bible verses.
+ */
+export class VerseEditorSuggestor extends EditorSuggest<VerseSuggesting> {
+  plugin: BibleReferencePlugin;
+  settings: BibleReferencePluginSettings;
+
+  constructor(
+    plugin: BibleReferencePlugin,
+    settings: BibleReferencePluginSettings
+  ) {
+    super(plugin.app);
+    this.plugin = plugin;
+    this.settings = settings;
+  }
+
+  /**
+   * This will build the EditorSuggestContext in getSuggestions
+   * @param cursor
+   * @param editor
+   * @param _
+   */
+  onTrigger(
+    cursor: EditorPosition,
+    editor: Editor,
+    _: TFile
+  ): EditorSuggestTriggerInfo | null {
+    const currentContent = editor.getLine(cursor.line).substring(0, cursor.ch);
+    const match = VerseTypoCheck(currentContent);
+    if (match) {
+      console.debug("trigger on", currentContent);
+      return {
+        end: cursor,
+        start: {
+          line: cursor.line,
+          ch: currentContent.lastIndexOf(match),
+        },
+        query: match,
+      };
+    }
+    return null;
+  }
+
+  /**
+   * Suggest bible verses.
+   * @param context
+   */
+  async getSuggestions(
+    context: EditorSuggestContext
+  ): Promise<VerseSuggesting[]> {
+    return getSuggestionsFromQuery(context.query, this.settings);
+  }
+
+  renderSuggestion(suggestion: VerseSuggesting, el: HTMLElement): void {
+    suggestion.renderSuggestion(el);
```

Comment:
I moved the actual rendering of a suggestion to the suggestion itself so it could be shared between this and the modal.